packages/mui-material/src/FilledInput/FilledInput.js

```diff
@@ -140,13 +148,28 @@ const FilledInputInput = styled(InputBaseInput, {
   paddingRight: 12,
   paddingBottom: 8,
   paddingLeft: 12,
-  '&:-webkit-autofill': {
-    WebkitBoxShadow: theme.palette.mode === 'light' ? null : '0 0 0 100px #266798 inset',
-    WebkitTextFillColor: theme.palette.mode === 'light' ? null : '#fff',
-    caretColor: theme.palette.mode === 'light' ? null : '#fff',
-    borderTopLeftRadius: 'inherit',
-    borderTopRightRadius: 'inherit',
-  },
+  ...(!theme.vars && {
+    '&:-webkit-autofill': {
+      WebkitBoxShadow: theme.palette.mode === 'light' ? null : '0 0 0 100px #266798 inset',
+      WebkitTextFillColor: theme.palette.mode === 'light' ? null : '#fff',
+      caretColor: theme.palette.mode === 'light' ? null : '#fff',
+      borderTopLeftRadius: 'inherit',
+      borderTopRightRadius: 'inherit',
+    },
+  }),
+  ...(theme.vars && {
+    '&:-webkit-autofill': {
+      borderTopLeftRadius: 'inherit',
+      borderTopRightRadius: 'inherit',
+    },
+    [theme.getColorSchemeSelector('dark')]: {
+      '&:-webkit-autofill': {
+        WebkitBoxShadow: '0 0 0 100px #266798 inset',
+        WebkitTextFillColor: '#fff',
+        caretColor: '#fff',
+      },
+    },
+  }),
```

Comment:
@mnajdova This is the logic when CSS variables exist which uses CSS selector when the mode is dark. I think we can move forward with this (it does not cause dark mode flicker) instead of creating 3 new tokens (the more tokens the bigger CSS output).

Reply:
Ah that makes sense 👍